external/kafka-0-10-token-provider/src/main/scala/org/apache/spark/kafka010/KafkaTokenUtil.scala

```diff
@@ -291,13 +291,14 @@ private[spark] object KafkaTokenUtil extends Logging {
   }
 
   def isConnectorUsingCurrentToken(
+      sparkConf: SparkConf,
       params: ju.Map[String, Object],
       clusterConfig: Option[KafkaTokenClusterConf]): Boolean = {
-    if (params.containsKey(SaslConfigs.SASL_JAAS_CONFIG)) {
+    if (sparkConf.getBoolean("spark.security.credentials.kafka.enabled", true) &&
```

Comment:
I'm kinda wondering if this extra check is needed, instead of just `if (clusterConfig.isDefined && params.containsKey(SaslConfigs.SASL_JAAS_CONFIG)`.

If it's really needed it's ok, although I'm not a big fan of hardcoding that config name here (instead of trying to reuse `HadoopDelegationTokenManager.isServiceEnabled`).

On a side note this method would be better if called `needTokenUpdate`; it's kinda weird to return `true` (as in "yes it's using the current token") when there is no token at all.

Reply:
Requiring HadoopDelegationTokenManager instance seems to be overkill given we  only need `isServiceEnabled`.

In other words, `isServiceEnabled` doesn't seem to be necessarily to be the method of `HadoopDelegationTokenManager` class. I can't find any usage of overriding though it's scope is protected. IMO it would be better we can move the `isServiceEnabled` method to HadoopDelegationTokenManager `object` (with requiring spark conf as parameter) and reuse.

Reply:
> I'm kinda wondering if this extra check is needed, instead of just if (clusterConfig.isDefined && params.containsKey(SaslConfigs.SASL_JAAS_CONFIG).

If `spark.security.credentials.kafka.enabled` is `false`, the cluster is defined in the `SparkConf` the user can set its own `SASL_JAAS_CONFIG`. Such case delegation token would be assumed by the code which is wrong.

Reply:
`needTokenUpdate` is better name so renamed.

Reply:
I've gone through and `HadoopDelegationTokenManager.isServiceEnabled` is more safe to use so moved it to object level.